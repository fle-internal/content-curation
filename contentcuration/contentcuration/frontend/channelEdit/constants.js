import { fileErrors } from 'shared/constants';

export const RouterNames = {
  SANDBOX: 'SANDBOX',
  TREE_ROOT_VIEW: 'TREE_ROOT_VIEW',
  TREE_VIEW: 'TREE_VIEW',
  STAGING_TREE_VIEW: 'STAGING_TREE_VIEW',
  CONTENTNODE_DETAILS: 'CONTENTNODE_DETAILS',
  MULTI_CONTENTNODE_DETAILS: 'MULTI_CONTENTNODE_DETAILS',
  IMPORT_FROM_CHANNELS: 'IMPORT_FROM_CHANNELS',
  IMPORT_FROM_CHANNELS_BROWSE: 'IMPORT_FROM_CHANNELS_BROWSE',
  IMPORT_FROM_CHANNELS_SEARCH: 'IMPORT_FROM_CHANNELS_SEARCH',
  IMPORT_FROM_CHANNELS_REVIEW: 'IMPORT_FROM_CHANNELS_REVIEW',
  ADD_TOPICS: 'ADD_TOPICS',
  ADD_EXERCISE: 'ADD_EXERCISE',
  UPLOAD_FILES: 'UPLOAD_FILES',
  TRASH: 'TRASH',
  ADD_PREVIOUS_STEPS: 'ADD_PREVIOUS_STEPS',
  ADD_NEXT_STEPS: 'ADD_NEXT_STEPS',
};

export const viewModes = {
  DEFAULT: 'DEFAULT_VIEW',
  COMFORTABLE: 'COMFORTABLE_VIEW',
  COMPACT: 'COMPACT_VIEW',
};

export const ValidationErrors = {
  TITLE_REQUIRED: 'TITLE_REQUIRED',
  LICENCE_REQUIRED: 'LICENCE_REQUIRED',
  COPYRIGHT_HOLDER_REQUIRED: 'COPYRIGHT_HOLDER_REQUIRED',
  LICENCE_DESCRIPTION_REQUIRED: 'LICENCE_DESCRIPTION_REQUIRED',
  MASTERY_MODEL_REQUIRED: 'MASTERY_MODEL_REQUIRED',
  MASTERY_MODEL_INVALID: 'MASTERY_MODEL_INVALID',
  QUESTION_REQUIRED: 'QUESTION_REQUIRED',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
  NO_VALID_PRIMARY_FILES: 'NO_VALID_PRIMARY_FILES',
  ...fileErrors,
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemToolbarActions = {
  EDIT_ITEM: 'EDIT_ITEM',
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  DELETE_ITEM: 'DELETE_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'questionTypeSingleSelection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'questionTypeMultipleSelection',
  [AssessmentItemTypes.TRUE_FALSE]: 'questionTypeTrueFalse',
  [AssessmentItemTypes.INPUT_QUESTION]: 'questionTypeInput',
};

export const TabNames = {
  DETAILS: 'details',
  PREVIEW: 'preview',
  QUESTIONS: 'questions',
  RELATED: 'related',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};
