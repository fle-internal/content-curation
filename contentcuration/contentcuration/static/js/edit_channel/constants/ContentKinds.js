module.exports = [
  {
    kind: 'document',
  },
  {
    kind: 'topic',
  },
  {
    kind: 'video',
  },
  {
    kind: 'html5',
  },
  {
    kind: 'audio',
  },
  {
    kind: 'exercise',
  },
];
