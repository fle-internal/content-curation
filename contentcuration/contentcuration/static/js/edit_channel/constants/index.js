const ContentKinds = require('./ContentKinds');
const FormatPresets = require('./FormatPresets');
const Languages = require('./Languages');
const Licenses = require('./Licenses');
const MasteryModels = require('./MasteryModels');
const Roles = require('./Roles');

module.exports = {
  ContentKinds,
  FormatPresets,
  Languages,
  Licenses,
  MasteryModels,
  Roles,
};
