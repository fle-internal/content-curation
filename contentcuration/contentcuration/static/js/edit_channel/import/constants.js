export const PageTypes = {
  TREE_VIEW: 'TREE_VIEW',
  IMPORT_PREVIEW: 'IMPORT_PREVIEW',
  SEARCH_RESULTS: 'SEARCH_RESULTS',
};
