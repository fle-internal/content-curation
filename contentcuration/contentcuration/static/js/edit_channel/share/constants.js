export const Permissions = {
  OWNER: 'manage',
  EDIT: 'edit',
  VIEW_ONLY: 'view'
};

export const PermissionRanks = [
	{
		"shareMode": Permissions.VIEW_ONLY,
		"rank": 0,
		"field": 'viewers'
	},
	{
		"shareMode": Permissions.EDIT,
		"rank": 1,
		"field": 'editors'
	},
	{
		"shareMode": Permissions.OWNER,
		"rank": 2,
		"field": null
	},
]
