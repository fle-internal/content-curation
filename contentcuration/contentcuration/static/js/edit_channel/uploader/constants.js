export const TabNames = {
  DETAILS: 'DETAILS',
  PREVIEW: 'PREVIEW',
  QUESTIONS: 'QUESTIONS',
  PREREQUISITES: 'PREREQUISITES',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};

export const AssessmentItemToolbarActions = {
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  EDIT_ITEM: 'EDIT_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
  DELETE_ITEM: 'DELETE_ITEM',
};

export const AssessmentItemValidationErrors = {
  BLANK_QUESTION: 'BLANK_QUESTION',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'Single selection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'Multiple selection',
  [AssessmentItemTypes.TRUE_FALSE]: 'True/False',
  [AssessmentItemTypes.INPUT_QUESTION]: 'Input question',
};
