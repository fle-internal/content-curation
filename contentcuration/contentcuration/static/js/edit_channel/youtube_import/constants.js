export const PageTypes = {
  SUBMIT_URL: 'SUBMIT_URL',
  CONFIRM: 'CONFIRM',
};

export const ImportStatus = {
  IDLE: 'IDLE',
  EXTRACTING: 'EXTRACTING',
  EXTRACT_ERROR: 'EXTRACT_ERROR',
  DOWNLOADING: 'DOWNLOADING',
  DOWNLOAD_ERROR: 'DOWNLOAD_ERROR',
  START_IMPORT: 'START_IMPORT',
  DONE: 'DONE',
};
