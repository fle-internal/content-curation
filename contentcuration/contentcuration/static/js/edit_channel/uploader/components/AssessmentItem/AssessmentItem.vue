<template>
  <VExpansionPanelContent>
    <template v-slot:header>
      <template v-if="!isOpen">
        <VFlex xs1>
          {{ order }}
        </VFlex>
        <VFlex>
          <div class="caption grey--text mb-1">
            {{ kindLabel }}
          </div>
          <div data-test="questionText">
            {{ question }}
          </div>
        </VFlex>
      </template>

      <template v-else>
        <VFlex xs12>
          {{ order }}
        </VFlex>
      </template>
    </template>

    <VCard>
      <VContainer fluid>
        <VLayout row>
          <VFlex xs5>
            <VSelect
              :items="kindSelectItems"
              :value="kind"
              label="Question type"
              data-test="kindSelect"
              @input="onKindChange"
            />
          </VFlex>
        </VLayout>

        <VLayout row>
          <VFlex>
            <VTextField
              label="Question text"
              :value="question"
              @input="onQuestionChange"
            />
          </VFlex>
        </VLayout>

        <!-- TODO @MisRob: Find out which linter tool is removing
        dashes from Vuetify attributes and disable -->
        <!-- eslint-disable-next-line -->
        <VLayout row mt-3>
          <div class="grey--text text--darken-1 mb-3">
            Answers
          </div>
        </VLayout>

        <!-- eslint-disable-next-line -->
        <VLayout row mb-3>
          <VBtn flat>
            New answer
          </VBtn>
        </VLayout>

        <VDivider />

        <!-- eslint-disable-next-line -->
        <VLayout row mt-3>
          <div class="grey--text text--darken-1 mb-3">
            Hints
          </div>
        </VLayout>

        <VLayout row>
          <VBtn flat>
            New hint
          </VBtn>
        </VLayout>

        <!-- eslint-disable-next-line -->
        <VLayout row justify-end>
          <VBtn
            flat
            color="primary"
            data-test="closeBtn"
            @click="onCloseClick"
          >
            Close
          </VBtn>
        </VLayout>
      </VContainer>
    </VCard>
  </VExpansionPanelContent>
</template>

<script>

  import { AssessmentItemTypes } from '../../constants';

  export default {
    name: 'AssessmentItem',
    props: {
      item: {
        type: Object,
      },
      itemIdx: {
        type: Number,
      },
      isOpen: {
        type: Boolean,
        default: false,
      },
    },
    computed: {
      order() {
        if (!this.item || this.item.order === undefined) {
          return 1;
        }

        return this.item.order + 1;
      },
      question() {
        if (!this.item || !this.item.question) {
          return '';
        }

        return this.item.question;
      },
      kind() {
        if (!this.item || !this.item.type) {
          return AssessmentItemTypes.SINGLE_SELECTION;
        }

        return this.item.type;
      },
      kindSelectItems() {
        return [
          { value: AssessmentItemTypes.SINGLE_SELECTION, text: 'Single selection' },
          { value: AssessmentItemTypes.MULTIPLE_SELECTION, text: 'Multiple selection' },
          { value: AssessmentItemTypes.INPUT_QUESTION, text: 'Input question' },
          { value: AssessmentItemTypes.TRUE_FALSE, text: 'True/False' },
        ];
      },
      kindLabel() {
        return this.kindSelectItems.find(item => item.value === this.kind).text;
      },
    },
    methods: {
      onQuestionChange(newQuestion) {
        this.$emit('update', {
          itemIdx: this.itemIdx,
          payload: { question: newQuestion },
        });
      },
      onKindChange(newKind) {
        this.$emit('update', { itemIdx: this.itemIdx, payload: { type: newKind } });
      },
      onCloseClick() {
        this.$emit('close');
      },
    },
  };

</script>
