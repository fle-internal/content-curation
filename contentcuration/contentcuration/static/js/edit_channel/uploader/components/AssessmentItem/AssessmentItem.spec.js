import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import { AssessmentItemTypes } from '../../constants';
import AssessmentItem from './AssessmentItem';

// TODO @MisRob: Consistent imports
const editModalGetters = require('../../vuex/getters');
const editModalMutations = require('../../vuex/mutations');

const localVue = createLocalVue();
localVue.use(Vuex);

const NODE_ID = 'exercise-2';
const ITEM_IDX = 1;
const ITEM = {
  id: 'exercise-2-item-2',
  question: 'Exercise 2 - Question 2',
  type: AssessmentItemTypes.SINGLE_SELECTION,
  order: 1,
  answers: [
    { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
    { answer: 'Peanut butter', correct: false, order: 2 },
  ],
  hints: [{ hint: "It's not healthy", order: 1 }, { hint: 'Tasty!', order: 2 }],
};

const EDIT_MODAL_STATE = {
  nodesAssessmentDrafts: {
    [NODE_ID]: [
      {
        id: 'exercise-2-item-1',
        question: 'Exercise 2 - Question 1',
        type: AssessmentItemTypes.INPUT_QUESTION,
        order: 0,
        answers: [
          { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
          { answer: 'Peanut butter', correct: true, order: 2 },
        ],
        hints: [],
      },
      ITEM,
      {
        id: 'exercise-2-item-3',
        question: 'Exercise 2 - Question 3',
        type: AssessmentItemTypes.INPUT_QUESTION,
        order: 2,
        answers: [],
        hints: [],
      },
    ],
  },
};

const clickCloseBtn = wrapper => {
  wrapper.find('[data-test=closeBtn]').trigger('click');
};

const selectKind = (wrapper, kind) => {
  const input = wrapper.find('[data-test=kindSelect]');
  input.element.value = kind;

  input.trigger('input');
};

const clickToolbarEditIcon = wrapper => {
  wrapper.find('[data-test=toolbarIconEdit]').trigger('click');
};

const updateQuestion = (wrapper, newQuestion) => {
  wrapper.find('[data-test=questionInput]').setValue(newQuestion);
};

const clickDeleteItem = wrapper => {
  wrapper.find('[data-test=toolbarMenuDeleteItem]').trigger('click');
};

const clickAddQuestionAbove = wrapper => {
  wrapper.find('[data-test=toolbarMenuAddItemAbove]').trigger('click');
};

const clickAddQuestionBelow = wrapper => {
  wrapper.find('[data-test=toolbarMenuAddItemBelow]').trigger('click');
};

const initWrapper = (state, propsData) => {
  const store = new Vuex.Store({
    modules: {
      edit_modal: {
        namespaced: true,
        state,
        getters: editModalGetters,
        mutations: editModalMutations,
      },
    },
  });

  return mount(AssessmentItem, {
    localVue,
    store,
    propsData,
  });
};

describe('AssessmentItem', () => {
  let wrapper;

  describe('when closed', () => {
    beforeEach(() => {
      const state = JSON.parse(JSON.stringify(EDIT_MODAL_STATE));
      const propsData = {
        nodeId: NODE_ID,
        itemIdx: ITEM_IDX,
        isOpen: false,
      };

      wrapper = initWrapper(state, propsData);
    });

    it('renders', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe('on toolbar edit click', () => {
      it('emits open event', () => {
        clickToolbarEditIcon(wrapper);

        expect(wrapper.emitted().open).toBeTruthy();
        expect(wrapper.emitted().open.length).toBe(1);
      });
    });
  });

  describe('when closed with answers preview', () => {
    beforeEach(() => {
      const state = JSON.parse(JSON.stringify(EDIT_MODAL_STATE));
      const propsData = {
        nodeId: NODE_ID,
        itemIdx: ITEM_IDX,
        isOpen: false,
        displayAnswersPreview: true,
      };

      wrapper = initWrapper(state, propsData);
    });

    it('renders', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('when open', () => {
    beforeEach(() => {
      const state = JSON.parse(JSON.stringify(EDIT_MODAL_STATE));
      const propsData = {
        nodeId: NODE_ID,
        itemIdx: ITEM_IDX,
        isOpen: true,
      };

      wrapper = initWrapper(state, propsData);
    });

    it('renders', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe('on item type update', () => {
      it('updates a correct item in drafts store', () => {
        selectKind(wrapper, AssessmentItemTypes.MULTIPLE_SELECTION);

        expect(
          wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID][ITEM_IDX]
        ).toEqual({ ...ITEM, type: AssessmentItemTypes.MULTIPLE_SELECTION });
      });
    });

    describe('on question update', () => {
      it('updates a correct item in drafts store', () => {
        updateQuestion(wrapper, 'What color is your eyes?');

        expect(
          wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID][ITEM_IDX]
        ).toEqual({ ...ITEM, question: 'What color is your eyes?' });
      });
    });

    describe('on answers update', () => {
      it('updates a correct item in drafts store', () => {
        const newAnswers = [
          { answer: 'Answer 1', correct: false, order: 1 },
          { answer: 'Answer 2', correct: true, order: 2 },
        ];

        wrapper.find({ name: 'AnswersEditor' }).vm.$emit('update', newAnswers);

        expect(
          wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID][ITEM_IDX]
        ).toEqual({
          ...ITEM,
          answers: [
            { answer: 'Answer 1', correct: false, order: 1 },
            { answer: 'Answer 2', correct: true, order: 2 },
          ],
        });
      });
    });

    describe('on hints update', () => {
      it('updates a correct item in drafts store', () => {
        const newHints = [{ hint: 'Hint 1', order: 1 }, { hint: 'Hint 2', order: 2 }];

        wrapper.find({ name: 'HintsEditor' }).vm.$emit('update', newHints);

        expect(
          wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID][ITEM_IDX]
        ).toEqual({ ...ITEM, hints: [{ hint: 'Hint 1', order: 1 }, { hint: 'Hint 2', order: 2 }] });
      });
    });
  });

  describe('on close click', () => {
    it('emits close event', () => {
      clickCloseBtn(wrapper);

      expect(wrapper.emitted().close).toBeTruthy();
      expect(wrapper.emitted().close.length).toBe(1);
    });
  });

  describe('on delete item click', () => {
    beforeEach(() => {
      const state = JSON.parse(JSON.stringify(EDIT_MODAL_STATE));
      const propsData = {
        nodeId: NODE_ID,
        itemIdx: ITEM_IDX,
      };

      wrapper = initWrapper(state, propsData);
      clickDeleteItem(wrapper);
    });

    it('removes item from drafts store', () => {
      expect(wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID]).toEqual([
        {
          id: 'exercise-2-item-1',
          question: 'Exercise 2 - Question 1',
          type: AssessmentItemTypes.INPUT_QUESTION,
          order: 0,
          answers: [
            { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
            { answer: 'Peanut butter', correct: true, order: 2 },
          ],
          hints: [],
        },
        {
          id: 'exercise-2-item-3',
          question: 'Exercise 2 - Question 3',
          type: AssessmentItemTypes.INPUT_QUESTION,
          order: 1,
          answers: [],
          hints: [],
        },
      ]);
    });
  });

  describe('on "Add question above" click', () => {
    beforeEach(() => {
      const state = JSON.parse(JSON.stringify(EDIT_MODAL_STATE));
      const propsData = {
        nodeId: NODE_ID,
        itemIdx: ITEM_IDX,
      };

      wrapper = initWrapper(state, propsData);
      clickAddQuestionAbove(wrapper);
    });

    it('adds a new item in drafts store before this item', () => {
      expect(wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID]).toEqual([
        {
          id: 'exercise-2-item-1',
          question: 'Exercise 2 - Question 1',
          type: AssessmentItemTypes.INPUT_QUESTION,
          order: 0,
          answers: [
            { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
            { answer: 'Peanut butter', correct: true, order: 2 },
          ],
          hints: [],
        },
        {
          question: '',
          type: AssessmentItemTypes.SINGLE_SELECTION,
          order: 1,
        },
        { ...ITEM, order: 2 },
        {
          id: 'exercise-2-item-3',
          question: 'Exercise 2 - Question 3',
          type: AssessmentItemTypes.INPUT_QUESTION,
          order: 3,
          answers: [],
          hints: [],
        },
      ]);
    });

    it('emits new item added event with an index of a newly added item', () => {
      expect(wrapper.emitted().newItemAdded).toBeTruthy();
      expect(wrapper.emitted().newItemAdded.length).toBe(1);
      expect(wrapper.emitted().newItemAdded[0][0]).toBe(1);
    });
  });

  describe('on "Add question below" click', () => {
    beforeEach(() => {
      const state = JSON.parse(JSON.stringify(EDIT_MODAL_STATE));
      const propsData = {
        nodeId: NODE_ID,
        itemIdx: ITEM_IDX,
      };

      wrapper = initWrapper(state, propsData);
      clickAddQuestionBelow(wrapper);
    });

    it('adds a new item in drafts store after this item', () => {
      expect(wrapper.vm.$store.state['edit_modal'].nodesAssessmentDrafts[NODE_ID]).toEqual([
        {
          id: 'exercise-2-item-1',
          question: 'Exercise 2 - Question 1',
          type: AssessmentItemTypes.INPUT_QUESTION,
          order: 0,
          answers: [
            { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
            { answer: 'Peanut butter', correct: true, order: 2 },
          ],
          hints: [],
        },
        ITEM,
        {
          question: '',
          type: AssessmentItemTypes.SINGLE_SELECTION,
          order: 2,
        },
        {
          id: 'exercise-2-item-3',
          question: 'Exercise 2 - Question 3',
          type: AssessmentItemTypes.INPUT_QUESTION,
          order: 3,
          answers: [],
          hints: [],
        },
      ]);
    });

    it('emits new item added event with an index of a newly added item', () => {
      expect(wrapper.emitted().newItemAdded).toBeTruthy();
      expect(wrapper.emitted().newItemAdded.length).toBe(1);
      expect(wrapper.emitted().newItemAdded[0][0]).toBe(2);
    });
  });
});
