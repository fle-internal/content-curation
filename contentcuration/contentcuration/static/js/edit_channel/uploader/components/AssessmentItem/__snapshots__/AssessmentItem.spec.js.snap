// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentItem when closed renders a correct order, type and text 1`] = `
<div class="v-card v-sheet theme--light">
  <div class="v-card__title header">
    <div class="container fluid">
      <div class="layout row">
        <div class="flex xs1">
          2
        </div>
        <div class="flex">
          <div class="caption grey--text mb-1">
            Single selection
          </div>
          <div data-test="questionText">
            What color is the sky?
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`AssessmentItem when open renders a correct order, type select, text input, answers editor, hints editor, close button 1`] = `
<div class="v-card v-sheet theme--light">
  <div data-test="open" class="v-card__title">
    <div class="container fluid">
      <div class="layout row">
        <div class="flex xs1">
          2
        </div>
        <div class="flex xs5">
          <div role="combobox" class="v-input v-text-field v-select v-input--is-label-active v-input--is-dirty theme--light">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-select__slot"><label aria-hidden="true" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Question type</label>
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">Single selection</div><input aria-label="Question type" data-test="kindSelect" readonly="readonly" type="text" autocomplete="on" aria-readonly="false">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_down</i></div>
                  </div>
                </div>
                <div class="v-menu">
                  <div class="v-menu__content theme--light " style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
                    <div class="v-select-list v-card theme--light">
                      <div role="list" class="v-list theme--light">
                        <div role="listitem" class="primary--text"><a class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">Single selection</div>
                            </div>
                          </a></div>
                        <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">Multiple selection</div>
                            </div>
                          </a></div>
                        <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">Input question</div>
                            </div>
                          </a></div>
                        <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">True/False</div>
                            </div>
                          </a></div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <div class="v-messages__wrapper"></div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="layout row justify-end">
        <div class="flex xs11">
          <div class="v-input v-text-field v-input--is-label-active v-input--is-dirty theme--light">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label aria-hidden="true" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Question text</label><input aria-label="Question text" data-test="questionInput" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <div class="v-messages__wrapper"></div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-card__text">
    <div class="container fluid">
      <div class="layout row justify-end">
        <div class="flex xs11">
          <div>
            <div class="grey--text text--darken-1 mb-3">
              Answers
            </div>
            <!---->
            <div class="v-card v-sheet theme--light">
              <div class="indicator wrong"></div>
              <div class="v-card__text">
                <div class="layout">
                  <div class="flex xs11">
                    <div class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light">
                      <div class="v-input__control">
                        <div class="v-input__slot" style="height: auto;">
                          <div role="radiogroup" class="v-input--radio-group__input">
                            <div class="v-radio theme--light">
                              <div class="v-input--selection-controls__input"><input aria-label="Yellow" aria-checked="false" role="radio" type="radio" name="v-radio-28" value="0" data-test="answerRadio">
                                <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                              </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Yellow</label>
                            </div>
                          </div>
                        </div>
                        <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <div data-test="answerToggle"><i aria-hidden="true" data-test="iconArrowDown" class="v-icon v-icon--link material-icons theme--light">keyboard_arrow_down</i></div>
                </div>
              </div>
            </div>
            <div class="v-card v-sheet theme--light">
              <div class="indicator wrong"></div>
              <div class="v-card__text">
                <div class="layout">
                  <div class="flex xs11">
                    <div class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light">
                      <div class="v-input__control">
                        <div class="v-input__slot" style="height: auto;">
                          <div role="radiogroup" class="v-input--radio-group__input">
                            <div class="v-radio theme--light">
                              <div class="v-input--selection-controls__input"><input aria-label="Black" aria-checked="false" role="radio" type="radio" name="v-radio-36" value="1" data-test="answerRadio">
                                <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                              </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Black</label>
                            </div>
                          </div>
                        </div>
                        <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <div data-test="answerToggle"><i aria-hidden="true" data-test="iconArrowDown" class="v-icon v-icon--link material-icons theme--light">keyboard_arrow_down</i></div>
                </div>
              </div>
            </div>
            <div class="v-card v-sheet theme--light">
              <div class="indicator correct"></div>
              <div class="v-card__text">
                <div class="layout">
                  <div class="flex xs11">
                    <div class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light">
                      <div class="v-input__control">
                        <div class="v-input__slot" style="height: auto;">
                          <div role="radiogroup" class="v-input--radio-group__input">
                            <div class="v-radio theme--light accent--text">
                              <div class="v-input--selection-controls__input"><input aria-label="Blue" aria-checked="true" role="radio" type="radio" name="v-radio-44" value="2" data-test="answerRadio">
                                <div class="v-input--selection-controls__ripple accent--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light accent--text">radio_button_checked</i>
                              </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">Blue</label>
                            </div>
                          </div>
                        </div>
                        <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <div data-test="answerToggle"><i aria-hidden="true" data-test="iconArrowDown" class="v-icon v-icon--link material-icons theme--light">keyboard_arrow_down</i></div>
                </div>
              </div>
            </div> <button type="button" class="v-btn v-btn--flat theme--light" data-test="newAnswerBtn">
              <div class="v-btn__content">
                New answer
              </div>
            </button>
          </div>
          <hr class="mt-3 mb-3 v-divider theme--light">
          <div>
            <div class="grey--text text--darken-1 mb-3">
              Hints
            </div>
            <!---->
            <div class="v-card v-sheet theme--light">
              <div class="v-card__text">
                <div class="layout row">
                  <div class="flex xs1">
                    1
                  </div>
                  <div class="flex xs10">
                    <p>
                      First hint
                    </p>
                  </div>
                  <div class="flex">
                    <div data-test="hintToggle"><i aria-hidden="true" data-test="iconArrowDown" class="v-icon v-icon--link material-icons theme--light">keyboard_arrow_down</i></div>
                  </div>
                </div>
              </div>
            </div>
            <div class="v-card v-sheet theme--light">
              <div class="v-card__text">
                <div class="layout row">
                  <div class="flex xs1">
                    2
                  </div>
                  <div class="flex xs10">
                    <p>
                      Second hint
                    </p>
                  </div>
                  <div class="flex">
                    <div data-test="hintToggle"><i aria-hidden="true" data-test="iconArrowDown" class="v-icon v-icon--link material-icons theme--light">keyboard_arrow_down</i></div>
                  </div>
                </div>
              </div>
            </div> <button type="button" class="v-btn v-btn--flat theme--light" data-test="newHintBtn">
              <div class="v-btn__content">
                New hint
              </div>
            </button>
          </div>
        </div>
      </div>
      <div class="layout row justify-end"><button type="button" class="v-btn v-btn--flat theme--light primary--text" data-test="closeBtn">
          <div class="v-btn__content">
            Close
          </div>
        </button></div>
    </div>
  </div>
</div>
`;
